test/clj/game/cards/programs_test.clj

```diff
@@ -1438,6 +1438,22 @@
           (card-ability state :runner (refresh davinci) 0)
           (click-card state :runner "The Turning Wheel"))
         (is (get-resource state 0) "The Turning Wheel is installed")
+        (is (find-card "DaVinci" (:discard (get-runner))) "DaVinci is trashed"))))
+  (testing "Using ability should trigger trash effects first. Issue #4987"
+    (do-game
+      (new-game {:runner {:id "Armand \"Geist\" Walker: Tech Lord"
+                          :deck ["Simulchip"]
+                          :hand ["DaVinci" "The Turning Wheel"]}})
```

Comment:
Is The Turning Wheel needed in this test?

Reply:
It's need. You have to have some valid target for DaVinci in hand. But you can then install new card (Simulchip) that has been drawn from deck via Geist ability.

Reply:
Cool, thank you